### ISSUE TITLE
ESM issues when running tests in Jest

### ISSUE THREAD

**Issue by undsoft** [marnusw/date-fns-tz#150]
Okay, so I've recently updated my Angular installation from 12 to 13. One of the things that changed as far as I can understand is greater use of ESM modules. The problem is that Jest doesn't yet fully support them, so `jest-preset-angular` provides a translation layer to make tests work: https://thymikee.github.io/jest-preset-angular/docs/guides/angular-13+

The only package I have problem with is `date-fns-tz`.

```
 /Users/undsoft/projects/work/test/webui/node_modules/date-fns/esm/format/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import isValid from "../isValid/index.js";
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      at Runtime.createScriptFromCode (../../../../.nvm/versions/node/v14.17.1/lib/node_modules/jest/node_modules/jest-runtime/build/index.js:1479:14)
      at Object.<anonymous> (node_modules/date-fns-tz/format/index.js:8:38)
```

I've tried using `testPathIgnorePatterns` in my `jest.config.js` and putting different values like: `node_modules/(?!(date-fns-tz/esm))`, `node_modules/(?!(date-fns-tz))`, `node_modules/(?!(date-fns))` and combinations. I've also unsuccessfully tried to use `moduleNameMapper` to map `date-fns-tz/esm` to `date-fns-tz`.

What did work was:
1. Opening up `node_modules/date-fns-tz/format/index.js` 
2. Changing
```
var _format = _interopRequireDefault(require("date-fns/format"));
```
to
```
var _format = _interopRequireDefault(require("date-fns").format);
```
3. And deleting `node_modules/date-fns-tz/esm` folder.
Doing this seems to help somehow.

## Reproduction
1. `git clone https://github.com/truenas/webui.git -b NAS-113293-reproduction`
2. `cd webui`
3. `yarn`
4. `./setup_env.js -i http://127.0.0.1/`
5. `jest --runTestsByPath ./src/app/core/components/pipes/format-datetime.pipe.spec.ts`

I will appreciate ideas on how to fix this. I don't know whether the problem is with `date-fns-tz`, `jest-preset-angular` or my setup.

**Comment by undsoft** [comment]
Similar issue was reported here: https://github.com/date-fns/date-fns/issues/2655, but it seemed to have been fixed by using different imports for `date-fns-tz`. In my case imports look okay to me:
```
import { format, utcToZonedTime } from 'date-fns-tz';
```

**Comment by marnusw** [comment]
When I have time I will convert the project to TypeScript and use a proper modern build step to generate ESM and CJS outputs with types. It seems like you have found a solution, so until then I will just leave this issue open.

**Comment by undsoft** [comment]
I wouldn't say I have a solution. My "solution" is patching `date-fns-tz` source code and deleting folders in `node_modules` :)

**Comment by undsoft** [comment]
If you could describe a list of steps of what you wanted to do, perhaps I could do a PR. This is really blocking us.

**Comment by marnusw** [comment]
If you can figure out the quicker way to solve this that will work we can go with that in the interin; it doesn't even have to be non-breaking, we could do a major release. 

But here is a rough list of what I would like to do:

- Convert the main and fp source functions to TypeScript
- Switch to an ES Module project, i.e. `"type": "module"`
- Set up a proper build process, I was thinking `tsdx` might be a good option to keep it simple
- Create esm, cjs and umd builds
- Use the [exports](https://nodejs.org/api/packages.html#exports) field in `package.json` to allow consuming esm/cjs rather than different import paths

If you're able to help with any of this it would be a massive contribution.

**Comment by tietzr** [comment]
I've ran into the same issue with Angular 13. I've tried a couple of changes but none seems to serve as a workaround.

**Comment by tietzr** [comment]
> What did work was:
> 
> 1. Opening up `node_modules/date-fns-tz/format/index.js`
> 2. Changing
> 
> ```
> var _format = _interopRequireDefault(require("date-fns/format"));
> ```
> 
> to
> 
> ```
> var _format = _interopRequireDefault(require("date-fns").format);
> ```
> 
> 3. And deleting `node_modules/date-fns-tz/esm` folder.
>    Doing this seems to help somehow.

Doing this also made the tests run successfully.

**Comment by marnusw** [comment]
Please see whether `1.3.0-beta.0` solves this for you.

**Comment by marnusw** [comment]
Fixed in `v1.3.0`
